src/main/java/org/reaktivity/specification/amqp/internal/AmqpFunctions.java

```diff
@@ -0,0 +1,391 @@
+ */
+package org.reaktivity.specification.amqp.internal;
+
+import org.agrona.MutableDirectBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.kaazing.k3po.lang.el.Function;
+import org.kaazing.k3po.lang.el.spi.FunctionMapperSpi;
+import org.reaktivity.specification.amqp.internal.types.*;
+import org.reaktivity.specification.amqp.internal.types.control.AmqpRouteExFW;
+import org.reaktivity.specification.amqp.internal.types.stream.AmqpAbortExFW;
+import org.reaktivity.specification.amqp.internal.types.stream.AmqpBeginExFW;
+import org.reaktivity.specification.amqp.internal.types.stream.AmqpDataExFW;
+
+import java.nio.charset.StandardCharsets;
+import java.util.function.Consumer;
+
+public final class AmqpFunctions
+{
+    private static final int MAX_BUFFER_SIZE = 1024 * 8;
+
+    public static class AmqpRouteExHelper
+    {
+        private final AmqpRouteExFW.Builder amqpRouteExRW;
+
+        public AmqpRouteExHelper()
+        {
+            MutableDirectBuffer writeBuffer = new UnsafeBuffer(new byte[MAX_BUFFER_SIZE]);
+            this.amqpRouteExRW = new AmqpRouteExFW.Builder()
+                .wrap(writeBuffer, 0, writeBuffer.capacity());
+        }
+
+        public AmqpRouteExHelper targetAddress(
+            String targetAddress)
+        {
+            amqpRouteExRW.targetAddress(targetAddress);
+            return this;
+        }
+
+        public AmqpRouteExHelper role(
+            String role)
+        {
+            amqpRouteExRW.role(
+                roleRW -> roleRW.set(AmqpRole.valueOf(role))
+            );
+            return this;
+        }
+
+        public byte[] build()
+        {
+            final AmqpRouteExFW amqpRouteEx = amqpRouteExRW.build();
+            final byte[] result = new byte[amqpRouteEx.sizeof()];
+            amqpRouteEx.buffer().getBytes(0, result);
+            return result;
+        }
+    }
+
+    public static class AmqpBeginExHelper
+    {
+        private final AmqpBeginExFW.Builder amqpBeginExRW;
+
+        public AmqpBeginExHelper()
+        {
+            MutableDirectBuffer writeBuffer = new UnsafeBuffer(new byte[MAX_BUFFER_SIZE]);
+            this.amqpBeginExRW = new AmqpBeginExFW.Builder()
+                .wrap(writeBuffer, 0, writeBuffer.capacity());
+        }
+
+        public AmqpBeginExHelper containerId(
+            String containerId)
+        {
+            amqpBeginExRW.containerId(containerId);
+            return this;
+        }
+
+        public AmqpBeginExHelper channel(
+            int channel)
+        {
+            amqpBeginExRW.channel(channel);
+            return this;
+        }
+
+        public AmqpBeginExHelper address(
+            String address)
+        {
+            amqpBeginExRW.address(address);
+            return this;
+        }
+
+        public AmqpBeginExHelper role(
+            String role)
+        {
+            amqpBeginExRW.role(
+                roleRW -> roleRW.set(AmqpRole.valueOf(role))
+            );
```

Comment:
Consider using shorter variable names in lamba expressions, allowing them to more often appear on a single line.
```
amqpBeginExRW.role(r -> r.set(AmqpRole.valueOf(role));
```
Also, consider renaming `amqpBeginExRW` to just `beginExRW` since we are in the `amqp` spec.